### PR TITLE
EWPP-1802: Improve live stream display.

### PR DESCRIPTION
**Pull request by yenyasinn**

### CODE REVIEW

File: modules/oe_content_event/config/install/field.field.node.oe_event.oe_event_online_link.yml [hunk 1]
```diff
@@ -18,5 +18,5 @@ default_value: {  }
 default_value_callback: ''
 settings:
   link_type: 17
-  title: 1
+  title: 2
```
Comment (yenyasinn) [inline modules/oe_content_event/config/install/field.field.node.oe_event.oe_event_online_link.yml:21]: From the spec
Screenshot 2021-12-07 at 20 51 15

File: modules/oe_content_event/src/EventNodeWrapper.php [hunk 2]
```diff
@@ -146,4 +146,70 @@ public function isRegistrationPeriodOver(\DateTimeInterface $datetime): bool {
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isOnlinePeriodOver(\DateTimeInterface $datetime): bool {
+    if (!$this->hasOnlineDates()) {
+      return TRUE;
+    }
+
+    return $this->hasOnlineDates() && $datetime >= $this->getOnlineEndDate()->getPhpDateTime();
```
Comment (sergepavle) [inline modules/oe_content_event/src/EventNodeWrapper.php:176]:
```suggestion
    return $datetime >= $this->getOnlineEndDate()->getPhpDateTime();
```

**Review by sergepavle — DISMISSED** [review]

File: modules/oe_content_event/src/EventNodeWrapper.php (hunk 2 above)
Comment (22Alexandra) [reply modules/oe_content_event/src/EventNodeWrapper.php:176]: I'd keep the check.

**Review by 22Alexandra — DISMISSED** [review]

**Review by nagyad — DISMISSED** [review]

**Review by sergepavle — DISMISSED** [review]

**Review by nagyad — DISMISSED** [review]

File: modules/oe_content_event/src/EventNodeWrapper.php [hunk 3]
```diff
@@ -146,4 +146,70 @@ public function isRegistrationPeriodOver(\DateTimeInterface $datetime): bool {
...
+   */
+  public function hasOnlineDates(): bool {
+    return !$this->entity->get('oe_event_online_dates')->isEmpty();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isOnlinePeriodYetToCome(\DateTimeInterface $datetime): bool {
+    if (!$this->hasOnlineDates()) {
```
Comment (imanoleguskiza) [inline modules/oe_content_event/src/EventNodeWrapper.php]: This check is not needed, we already check in the main return

File: modules/oe_content_event/src/EventNodeWrapper.php [hunk 4]
```diff
@@ -146,4 +146,70 @@ public function isRegistrationPeriodOver(\DateTimeInterface $datetime): bool {
...
+      && $this->getOnlineStartDate()->getPhpDateTime() <= $datetime
+      && $datetime < $this->getOnlineEndDate()->getPhpDateTime();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isOnlinePeriodOver(\DateTimeInterface $datetime): bool {
+    if (!$this->hasOnlineDates()) {
+      return TRUE;
```
Comment (imanoleguskiza) [inline modules/oe_content_event/src/EventNodeWrapper.php]: This is a bit misleadin, if there are no online dates its neither over nor it has not started. Lets follow the same logic as in isRegistrationPeriodOver.

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Review by nagyad — APPROVED** [review]

**Pull request merged**